pkg/apis/core/v1/storagecluster.go

```diff
@@ -387,6 +387,8 @@ type CloudStorageNodeSpec struct {
 
 // CloudStorageCommon details of storage in cloud environment
 type CloudStorageCommon struct {
+	// CloudProviderSpec spec for the cloud provider, such as GKE, AWS etc
+	CloudProviderSpec *string `json:"cloudProviderSpec,omitempty"`
```

Comment:
If this field goes in common spec, it also applies to node spec. I am not sure if we can support multiple cloud providers for a single cluster. Let's confirm that, else we can add this only for cluster level spec.